frontend/controller/actions/chatroom.js

```diff
@@ -1,17 +1,121 @@
 'use strict'
 import sbp from '@sbp/sbp'
+
+// Using relative path to crypto.js instead of ~-path to workaround some esbuild bug
+import { CURVE25519XSALSA20POLY1305, EDWARDS25519SHA512BATCH, keyId, keygen, serializeKey, encrypt } from '../../../shared/domains/chelonia/crypto.js'
 import { L, GIErrorUIRuntimeError } from '@common/common.js'
 import { omit } from '@model/contracts/shared/giLodash.js'
 import { encryptedAction } from './utils.js'
+import { GIMessage } from '~/shared/domains/chelonia/GIMessage.js'
 import type { GIRegParams } from './types.js'
 
 export default (sbp('sbp/selectors/register', {
   'gi.actions/chatroom/create': async function (params: GIRegParams) {
     try {
-      return await sbp('chelonia/out/registerContract', {
+      // Create the necessary keys to initialise the contract
+      // eslint-disable-next-line camelcase
+      const CSK = keygen(EDWARDS25519SHA512BATCH)
+      const CEK = keygen(CURVE25519XSALSA20POLY1305)
+
+      // Key IDs
+      const CSKid = keyId(CSK)
+      const CEKid = keyId(CEK)
+
+      // Public keys to be stored in the contract
+      const CSKp = serializeKey(CSK, false)
+      const CEKp = serializeKey(CEK, false)
+
+      // Secret keys to be stored encrypted in the contract
+      const CSKs = encrypt(CEK, serializeKey(CSK, true))
+      const CEKs = encrypt(CEK, serializeKey(CEK, true))
+
+      const rootState = sbp('state/vuex/state')
+
+      console.log('Chatroom create', {
```

Comment:
1. Debug logging should happen with `console.debug`, not `console.log`
2. Please log the variable instead of duplicating the value as a literal
3. "chatroom create" - can you improve this to be a human-readable (or at least semi-human readable) statement of what is being logged?

Reply:
This is meant for me during development and will be removed. The logging is already done by SBP,  but this helps me easily see some things as I develop.